intervals_parser.py

```diff
@@ -0,0 +1,130 @@
+        .withColumn('chrom', f.regexp_replace(f.split(f.col('location'), ':|-')[0], 'chr', ''))
+        .withColumn('start', f.split(f.col('location'), ':|-')[1].cast(t.IntegerType()))
+        .withColumn('end', f.split(f.col('location'), ':|-')[2].cast(t.IntegerType()))
+
+        # Select relevant columns:
+        .select('chrom', 'start', 'end', 'gene_symbol', 'score')
+
+        # Drop rows with non-canonical chromosomes:
+        .filter(f.col('chrom').isin([str(x) for x in range(1, 23)] + ['X', 'Y', 'MT']))
+
+        # For each region/gene, keep only one row with the highest score:
+        .groupBy('chrom', 'start', 'end', 'gene_symbol')
+        .agg(f.max('score').alias('score'))
+
+        .orderBy('chrom', 'start')
+        .persist()
+    )
+
+    # Prepare gene set:
+    genes = gene_index.withColumnRenamed('gene_name', 'gene_symbol').select('gene_symbol', 'chr', 'gene_id', 'TSS')
+
+    return (
+        # Lift over the intervals:
+        lift.convert_intervals(anderson_df, 'chrom', 'start', 'end')
+        .drop('start', 'end')
+        .withColumnRenamed('mapped_start', 'start')
+        .withColumnRenamed('mapped_end', 'end')
+        .distinct()
+
+        # Joining with the gene index (unfortunately we are losing a bunch of genes here due to old symbols):
+        .join(genes, on='gene_symbol', how='left')
+        .filter(
+            # Drop rows where the gene is not on the same chromosome
+            (f.col('chrom') == f.regexp_replace(f.col('chr'), 'chr', ''))
+            # Drop rows where the TSS is far from the start of the region
+            & (
+                f.abs(
+                    (f.col('start') + f.col('end')) / 2 - f.col('TSS')
+                ) <= proximity_limit
+            )
+        )
+
+        # Adding constant values:
+        .withColumn('dataset_name', f.lit(cfg.dataset_name))
+        .withColumn('data_type', f.lit(cfg.data_type))
+        .withColumn('experiment_type', f.lit(cfg.experiment_type))
+        .withColumn('pmid', f.lit(cfg.pubmed_id))
+        .withColumn('bio_feature', f.lit(cfg.bio_feature))
+
+        # Select relevant columns:
+        .select(
+            'chrom', 'start', 'end', 'gene_id', 'score', 'dataset_name', 'data_type', 'experiment_type', 'pmid', 'bio_feature'
+        )
+        .persist()
+    )
+
+@hydra.main(config_path='../configs', config_name='config')
+def main(cfg):
+
+    spark = (
+        pyspark.sql.SparkSession
+        .builder
+        .master("local[*]")
+        .getOrCreate()
+    )
+
+    chain_file = cfg.intervals.liftover_chain_file
+    max_difference = cfg.intervals.max_length_difference
+    proximity_limit = 2 * float(cfg.intervals.proximity_limit)
+
+    # Open and process gene file:
+    gene_index = spark.read.parquet(cfg.intervals.gene_index).persist()
+
+    # Initialize liftover object:
+    lift = LiftOverSpark(chain_file, max_difference)
+
+    # Parsing the anderson file:
```

Comment:
I'd suggest defining an array containing the different datasets like here https://github.com/opentargets/evidence_datasource_parsers/blob/215adb032afd1f23acac2e72973b2a6844301d90/modules/GeneBurden.py#L29